rpcs3/Emu/Cell/lv2/sys_hid.cpp

```diff
@@ -1,29 +1,103 @@
 #include "stdafx.h"
+#include "sys_hid.h"
+
 #include "Emu/Memory/vm.h"
+#include "Emu/System.h"
+#include "Emu/Cell/PPUThread.h"
+
 #include "Emu/Cell/ErrorCodes.h"
+#include "Emu/Io/PadHandler.h"
+#include "Emu/Cell/Modules/cellPad.h"
 
 #include "sys_process.h"
-#include "sys_hid.h"
 
 LOG_CHANNEL(sys_hid);
 
 error_code sys_hid_manager_open(u64 device_type, u64 port_no, vm::ptr<u32> handle)
 {
 	sys_hid.todo("sys_hid_manager_open(device_type=0x%llx, port_no=0x%llx, handle=*0x%llx)", device_type, port_no, handle);
 
+	//device type == 1 = pad, 2 = kb, 3 = mouse
+	if (device_type > 3)
+	{
+		return CELL_EINVAL;
+	}
+
+	// 'handle' starts at 0x100 in realhw, and increments every time sys_hid_manager_open is called
+	// however, sometimes the handle is reused when opening sys_hid_manager again (even when the previous one hasn't been closed yet) - maybe when processes/threads get killed/finish they also release their handles?
+	static u32 ctr = 0x100;
+	*handle = ctr++;
+
+	//const auto handler = fxm::import<pad_thread>(Emu.GetCallbacks().get_pad_handler);
+	cellPadInit(7);
+	cellPadSetPortSetting(0, 1 | 2);
```

Comment:
I'm not quite sure why you would call these two functions here.
The port setting is already set in cellPadInit. The only difference is that you set flag 1 here for whatever reason.
And if you really need to use it after all, you should at least use the constants and not some magic numbers

Reply:
Not sure why it was originally required for the flags, but the init call is required since the pad code is currently in hle cellpad, instead of sys_hid

Reply:
as I said before, there are constants for it.
Also, you should probably do this based on device_type